dask/dataframe/io/tests/test_csv.py

```diff
@@ -1287,6 +1287,61 @@ def test_to_csv_multiple_files_cornercases():
         assert_eq(result, df16)
 
 
+def test_to_single_csv():
+    df = pd.DataFrame({"x": ["a", "b", "c", "d"], "y": [1, 2, 3, 4]})
+
+    for npartitions in [1, 2]:
+        a = dd.from_pandas(df, npartitions)
+        with tmpdir() as dn:
+            fn = os.path.join(dn, "test.csv")
+            a.to_csv(fn, index=False, single_file=True)
+            result = dd.read_csv(fn).compute().reset_index(drop=True)
+            assert_eq(result, df)
+
+        with tmpdir() as dn:
+            fn = os.path.join(dn, "test.csv")
+            r = a.to_csv(fn, index=False, compute=False, single_file=True)
+            dask.compute(r, scheduler="sync")
+            result = dd.read_csv(fn).compute().reset_index(drop=True)
+            assert_eq(result, df)
+
+
+@pytest.mark.xfail(
```

Comment:
In my experience, these tests are written using `pytest.raises`, rather than an `xfail`.

```python
a = dd.from_pandas(df, 1)
with tmpdir() as dn:
    with pytest.raises(ValueError, match="name_function is not supported"):
        a.to_csv(...)
```